numpy/ma/core.py

```diff
@@ -3885,15 +3885,15 @@ def compress(self, condition, axis=None, out=None):
             _new._mask = _mask.compress(condition, axis=axis)
         return _new
 
-    def __str__(self):
+    def __str__(self, str_type=str):
```

Comment:
Though I'm also slightly wary of using different signatures for dunder methods. Before you know it, subclasses rely on it... Might it be an idea to make a new function `_prepare_str(self)` (better name surely possible...) which returns `res` without `str` applied and then use that in both `__str__` and `__unicode__`? Or possibly just the masked print option enabled part as `_fill_with_mask_for_str`?